node/src/signature_provider/ecdsa_secp256k1/kdf.rs

```diff
@@ -0,0 +1,195 @@
+use crate::hkdf::ScalarExt;
```

Comment:
:grimacing: This is a ton of moves and breaks history. Is it a lot of work for you to move these files back to the src directory and then follow up with another PR to move them?

Also nit on the package name - I'd suggest shortening, like signature_provider --> providers; ecdsa_secp256k1 -> secp256k1 or just ecdsa. These names do appear in tracing logs, and if it's too long it can make reading them less pleasant.

Reply:
> This is a ton of moves and breaks history. Is it a lot of work for you to move these files back to the src directory and then follow up with another PR to move them?

Will do

> Also nit on the package name

The reasoning behind was that a signature scheme is uniquely defined by the algorithm + its curve. But since we only have 1.5 providers, we can indeed shorten it without loss of distinguishability